### PR TITLE
Update scroll bar with scroll

### PR DESCRIPTION
**Pull request by Kapperchino**
## Summary of the Pull Request
Fixes the problem where the scrollbar doesn't update when scrolling with keyboard shortcuts
## References
#821

## PR Checklist
* [x] Closes #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Manually tested
* [x] Doesn't require documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #821

### CODE REVIEW

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -1047,6 +1047,17 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     {
         _terminal->UserScrollViewport(viewTop);
     }
+    // Method Description:
+    // - Scrolls the viewport of the terminal and updates the scroll bar accordingly
+    // Arguments:
+    // - viewTop: the viewTop to scroll to
+    void TermControl::KeyBoardScrollViewPort(int viewTop)
+    {
+        _terminal->UserScrollViewport(viewTop);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/TermControl.cpp]: You know, I think that all of this should honestly just be in `TermControl::ScrollViewport`. I think `ScrollViewport` is supposed to be the general purpose method for a user-initiated scroll event called on the control (as opposed to a scroll that was initiated by the terminal itself).

Did anything break by putting this in `ScrollViewport`?

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 2]
```diff
@@ -1047,6 +1047,17 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
...
         _terminal->UserScrollViewport(viewTop);
     }
+    // Method Description:
+    // - Scrolls the viewport of the terminal and updates the scroll bar accordingly
+    // Arguments:
+    // - viewTop: the viewTop to scroll to
+    void TermControl::KeyBoardScrollViewPort(int viewTop)
+    {
+        _terminal->UserScrollViewport(viewTop);
+        const auto currentOffset = this->GetScrollOffset();
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/TermControl.cpp]: This guy looks unused...

**Comment by zadjii-msft** [comment]
@msftbot make sure I sign off on this

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 1 above)
Comment (Kapperchino) [reply src/cascadia/TerminalControl/TermControl.cpp]: `ScrollViewport` was being called by `_ScrollbarChangeHandler`, it doesn't break anything but it makes the scroll bar less smooth and jittery when I put the changes in `ScrollViewport` instead

Comment (zadjii-msft) [reply src/cascadia/TerminalControl/TermControl.cpp]: Oh yea, that might just be right. Sounds good to me!

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 3]
```diff
@@ -1047,6 +1047,16 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     {
         _terminal->UserScrollViewport(viewTop);
     }
+    // Method Description:
+    // - Scrolls the viewport of the terminal and updates the scroll bar accordingly
+    // Arguments:
+    // - viewTop: the viewTop to scroll to
+    void TermControl::KeyBoardScrollViewPort(int viewTop)
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/TermControl.cpp]: _the smallest nit_: 
```suggestion
    void TermControl::KeyBoardScrollViewport(int viewTop)
```

and everywhere else - usually we use "Viewport" as one word

File: src/cascadia/TerminalApp/Tab.cpp [hunk 4]
```diff
@@ -87,6 +87,6 @@ void Tab::Scroll(int delta)
 {
     _control.GetControl().Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [=](){
         const auto currentOffset = _control.GetScrollOffset();
-        _control.ScrollViewport(currentOffset + delta);
+        _control.KeyBoardScrollViewPort(currentOffset + delta);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/Tab.cpp]:
```suggestion
        _control.KeyBoardScrollViewport(currentOffset + delta);
```

File: src/cascadia/TerminalControl/TermControl.h [hunk 5]
```diff
@@ -43,6 +43,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         void Close();
 
         void ScrollViewport(int viewTop);
+        void KeyBoardScrollViewPort(int viewTop);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/TermControl.h]:
```suggestion
        void KeyBoardScrollViewport(int viewTop);
```

File: src/cascadia/TerminalControl/TermControl.idl [hunk 6]
```diff
@@ -35,6 +35,7 @@ namespace Microsoft.Terminal.TerminalControl
         void Close();
 
         void ScrollViewport(Int32 viewTop);
+        void KeyBoardScrollViewPort(Int32 viewTop);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/TermControl.idl]:
```suggestion
        void KeyBoardScrollViewport(Int32 viewTop);
```

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 7]
```diff
@@ -1047,6 +1047,16 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     {
         _terminal->UserScrollViewport(viewTop);
     }
+    // Method Description:
+    // - Scrolls the viewport of the terminal and updates the scroll bar accordingly
+    // Arguments:
+    // - viewTop: the viewTop to scroll to
+    void TermControl::KeyBoardScrollViewport(int viewTop)
```
Comment (miniksa) [inline src/cascadia/TerminalControl/TermControl.cpp]: nit: I've never seen it written as Key Board, always Keyboard. 

```suggestion
    void TermControl::KeyboardScrollViewport(int viewTop)
```

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 1 above)
Comment (miniksa) [reply src/cascadia/TerminalControl/TermControl.cpp]: Can you please leave this research/insight behind in a comment so someone in the future doesn't get the bright idea to consolidate them without this knowledge?

File: src/cascadia/TerminalControl/TermControl.h [hunk 8]
```diff
@@ -43,6 +43,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         void Close();
 
         void ScrollViewport(int viewTop);
+        void KeyBoardScrollViewport(int viewTop);
```
Comment (miniksa) [inline src/cascadia/TerminalControl/TermControl.h]:
```suggestion
        void KeyboardScrollViewport(int viewTop);
```

File: src/cascadia/TerminalControl/TermControl.idl [hunk 9]
```diff
@@ -35,6 +35,7 @@ namespace Microsoft.Terminal.TerminalControl
         void Close();
 
         void ScrollViewport(Int32 viewTop);
+        void KeyBoardScrollViewport(Int32 viewTop);
```
Comment (miniksa) [inline src/cascadia/TerminalControl/TermControl.idl]:
```suggestion
        void KeyboardScrollViewport(Int32 viewTop);
```

File: src/cascadia/TerminalApp/Tab.cpp [hunk 10]
```diff
@@ -87,6 +87,6 @@ void Tab::Scroll(int delta)
 {
     _control.GetControl().Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [=](){
         const auto currentOffset = _control.GetScrollOffset();
-        _control.ScrollViewport(currentOffset + delta);
+        _control.KeyBoardScrollViewport(currentOffset + delta);
```
Comment (miniksa) [inline src/cascadia/TerminalApp/Tab.cpp]:
```suggestion
        _control.KeyboardScrollViewport(currentOffset + delta);
```

**Pull request merged**